client/components/themes-list/index.jsx

```diff
@@ -131,6 +131,7 @@ export const ThemesList = ( { tabFilter, ...props } ) => {
 			{ tabFilter !== 'my-themes' && props.themes.length > 0 && (
 				<PatternAssemblerCta onButtonClick={ goToSiteAssemblerFlow } />
 			) }
+			{ props.children }
```

Comment:
What's the reason for passing down all these children in `ThemesList` and `ThemesSelection`?

Reply:
There are multiple reasons for this approach, I'll do my best in trying to explain them.

By passing down children instead of prop drilling if Discovery is enabled and if we're in a specific view, we obtain a more cohesive component since the themes-list component isn't coupled to those specific properties from outside of its problem domain. By using this approach our architecture becomes more cohesive.

The alternative would be to either pass the necessary properties down the ThemeList and ThemeSelection components or to pass down a built component as a property (pretty similar to using children but in a prop instead) the main drawback of these approaches is that they don't scale as well. If we ever need to add a new component to the theme list we will need to modify more than one file for this. With the suggested implementation, if we ever need to add a new foreign element to the list we will only need to modify the Theme Showcase component, this makes the current proposal adhere better to the single responsibility principle, which in turn reduces overall complexity and increases maintainability.

If we were to change the implementation, the dependency graph would also get distributed across more components, right now we've only added a single dependency to the Theme Showcase.

Another thing about implementing it as the `PatternAssemblerCta` is that we're incrementing the scope of the domain of the theme-list component. Does the theme-list component really need to know how when and where to render a CTA or foreign elements? The problem it should be solving is to render theme card elements and maybe track clicks, anything besides that is scope creeping into its problem domain.

Reply:
Aight, I guess it's fine. 🙂 

> Another thing about implementing it as the PatternAssemblerCta is that we're incrementing the scope of the domain of the theme-list component. 

Yeah, I think this is more the Showcase's fault than anything else.
We have like 3 nested list components for no good reason.
I suggested doing the same as the CTA because we'd piggyback on an existing exception instead of [creating a new one](https://xkcd.com/927/). 🙂